modules/33acrossBidAdapter.js

```diff
@@ -137,62 +147,85 @@ function _validateVideo(bid) {
 // NOTE: With regards to gdrp consent data, the server will independently
 // infer the gdpr applicability therefore, setting the default value to false
 function buildRequests(bidRequests, bidderRequest) {
+  const ttxSettings = getTTXConfig();
+
   const gdprConsent = Object.assign({
     consentString: undefined,
     gdprApplies: false
   }, bidderRequest && bidderRequest.gdprConsent);
 
   const uspConsent = bidderRequest && bidderRequest.uspConsent;
+
   const pageUrl = (bidderRequest && bidderRequest.refererInfo) ? (bidderRequest.refererInfo.referer) : (undefined);
 
   adapterState.uniqueSiteIds = bidRequests.map(req => req.params.siteId).filter(utils.uniques);
 
-  return bidRequests.map(bidRequest => _createServerRequest(
-    {
-      bidRequest,
-      gdprConsent,
-      uspConsent,
-      pageUrl
-    })
-  );
+  const bidRequestsComplete = bidRequests.map(_inferProduct);
+
+  const enableSRAMode = ttxSettings && ttxSettings.enableSRAMode;
+
+  const keyFunc = (enableSRAMode === true) ? _getSRAKey : _getDefaultKey;
+  const groupedRequests = _groupBidRequests(bidRequestsComplete, keyFunc);
+
+  const serverRequests = [];
+
+  for (const key in groupedRequests) {
```

Comment:
This an optional recommendation, maybe the `buildRequests` function could be splitted in order to reduce its responsibilities.
Something like having a `buildGroupedBidRequests` and `buildServerRequests`

Reply:
Oh meaning split the responsibility of generating request groups away from `buildServerRequests` ?

Reply:
yes, in order to reduce the number of statements inside this function and make it easier to read.

Reply:
Done, see updates.